azure-monitor-opentelemetry/tests/configuration/test_configure.py

```diff
@@ -404,197 +404,194 @@ def test_setup_metrics(
         self,
-        getattr_mock,
+        iter_mock,
+        dep_mock,
         logger_mock,
     ):
-        with patch("importlib.import_module"):
-            instrument_mock = Mock()
-            instrumentor_mock = Mock()
-            instrumentor_mock.return_value = instrument_mock
-            side_effect_calls = []
-            for _ in _SUPPORTED_INSTRUMENTED_LIBRARIES:
-                side_effect_calls.append(Exception())
-            getattr_mock.side_effect = side_effect_calls
-            _setup_instrumentations({})
-            import_calls = []
-            for lib_name in _SUPPORTED_INSTRUMENTED_LIBRARIES:
-                import_calls.append(call(lib_name))
-                import_calls.append(
-                    call("opentelemetry.instrumentation." + lib_name)
-                )
-            instrumentor_mock.assert_not_called()
-            instrument_mock.instrument.assert_not_called()
+        configurations = {}
+        ep_mock = Mock()
+        iter_mock.return_value = (ep_mock,)
+        instrumentor_mock = Mock()
+        instr_class_mock = Mock()
+        instr_class_mock.return_value = instrumentor_mock
+        ep_mock.name = _SUPPORTED_INSTRUMENTED_LIBRARIES[0]
+        ep_mock.load.side_effect = Exception()
+        dep_mock.return_value = None
+        _setup_instrumentations(configurations)
+        dep_mock.assert_called_with(ep_mock.dist)
+        ep_mock.load.assert_called_once()
+        instrumentor_mock.instrument.assert_not_called()
+        logger_mock.warning.assert_called_once()
 
-    @patch("azure.monitor.opentelemetry._configure.getattr")
+    @patch(
+        "azure.monitor.opentelemetry._configure.get_dist_dependency_conflicts"
+    )
+    @patch("azure.monitor.opentelemetry._configure.iter_entry_points")
     def test_setup_instrumentations_custom_configuration(
         self,
-        getattr_mock,
+        iter_mock,
+        dep_mock,
     ):
-        with patch("importlib.import_module"):
-            config_libr_name = _SUPPORTED_INSTRUMENTED_LIBRARIES[0]
-            configurations = {
-                config_libr_name + "_config": {"test_key": "test_value"},
-            }
-            instrument_mock = Mock()
-            instrumentor_mock = Mock()
-            instrumentor_mock.return_value = instrument_mock
-            getattr_mock.return_value = instrumentor_mock
-            _setup_instrumentations(configurations)
-            instrument_calls = [{"test_key": "test_value"}]
-            for _ in _SUPPORTED_INSTRUMENTED_LIBRARIES[1:-1]:
-                instrument_calls.append(call())
+        libr_name = _SUPPORTED_INSTRUMENTED_LIBRARIES[0]
+        configurations = {
+            libr_name + "_config": {"test_key": "test_value"},
+        }
+        ep_mock = Mock()
+        iter_mock.return_value = [ep_mock]
+        instrumentor_mock = Mock()
+        instr_class_mock = Mock()
+        instr_class_mock.return_value = instrumentor_mock
```

Comment:
NIT: Lots of common setup for these tests. Consider putting them into a helper method to make it easier to read and to add tests.

Reply:
Personally I do not think common setup will help that much in these test cases as they each differ slightly in what they are testing. Having explicit mocks called out each time makes it easier to understand and compare between tests the slight differences to see what is being tested. Not a big opinion and could do another refactor PR if we really need to. wdyt?